libs/libsystem/malloc.h

```diff
@@ -0,0 +1,28 @@
+#ifndef __oneOS__libsystem__MALLOC_H
+#define __oneOS__libsystem__MALLOC_H
+
+#include "types.h"
+
+#define DEFAULT_BLOCK_SIZE 4096
+#define HEADER_SIZE 64
+#define MAX_ALLOCATED_BLOCKS 64
+
+typedef struct malloc_header {
+    size_t size;
+    bool free;
+    struct malloc_header* next;
+    struct malloc_header* prev;
+} malloc_header_t;
+
+void* memory[MAX_ALLOCATED_BLOCKS]; // keeps pointers to the memmory, allocated by mmap

```

Comment:
replace them with `extern void* memory[MAX_ALLOCATED_BLOCKS];` if other parts should have access to them. Or just put them in .c file